### ISSUE TITLE
prop_mutlinode_pruning_Sim failure

### ISSUE THREAD

**Issue by coot** [input-output-hk/ouroboros-network#3456]
See the `coot/prop_pruning_failure` branch for the test case.  It triggers:
```
Exception thrown while showing test case:
  Assertion failed
  CallStack (from HasCallStack):
    assert, called at src\Ouroboros\Network\ConnectionManager\Core.hs:702:17 in ouroboros-network-framework-0.1.0.0-inplace:Ouroboros.Network.ConnectionManager.Core
```

**Comment by coot** [comment]
I found another counter example:
```
prop :: Property
prop = prop_inbound_governor_pruning
  (-14)
  (MultiNodePruningScript (AcceptedConnectionsLimit {acceptedConnectionsHardLimit = 1, acceptedConnectionsSoftLimit = 0, acceptedConnectionsDelay = 0}) [StartServer 0.3 (TestAddr {unTestAddr = TestAddress 76}) 19,InboundConnection 0.06 (TestAddr {unTestAddr = TestAddress 76}),OutboundConnection 3 (TestAddr {unTestAddr = TestAddress 76}),InboundMiniprotocols 2.9 (TestAddr {unTestAddr = TestAddress 76}) (Bundle {withHot = WithHot [-25,-27,-2,12,9,12,8,30,3,-3,-39,-16,33,7,4,-25,-31,12,40,27,-4,27,-42,40,-3], withWarm = WithWarm [-25,11,-28,39,-13,16,26,-32,-35,41,24,35,-15,4,3,-13,26,29,36,9], withEstablished = WithEstablished [-28,27,-35,-8,17,25,-40,-2,27,33,27,-36,-15,34,-3,-7,-10,-1,2,-25,-45,-18,9,12,18,36]}),CloseInboundConnection 0.19 (TestAddr {unTestAddr = TestAddress 76}),StartServer 0.01 (TestAddr {unTestAddr = TestAddress 62}) (-13),InboundConnection 0.31 (TestAddr {unTestAddr = TestAddress 62}),OutboundConnection 0.23 (TestAddr {unTestAddr = TestAddress 62}),InboundMiniprotocols 4.1 (TestAddr {unTestAddr = TestAddress 62}) (Bundle {withHot = WithHot [45,2,47,-11,5,46,-30,-44,-10,-45,-44,24,46,39,2,-21,-31,-14,-36,3,45,36,-35,-46,11,38,2,-42,-7], withWarm = WithWarm [47,-13,11,-5,1,-14,-15,5,-22,-28,-28,3,27,-48,37,-45,-20,-6,-22,-2,20,39,-6,-8,-44,13,-3,33,32,45,-22,-19,-14,19,-13,-21,-44,-29,-8,-30,-19,-17,4,-41,31,37,37], withEstablished = WithEstablished [31,-28,40,-13,32,20,11,-29,39,-38,14,-28]}),CloseOutboundConnection 2.2 (TestAddr {unTestAddr = TestAddress 76}),StartServer 0.011 (TestAddr {unTestAddr = TestAddress 13}) 17,InboundConnection 0.022 (TestAddr {unTestAddr = TestAddress 13}),OutboundConnection 0.046 (TestAddr {unTestAddr = TestAddress 13}),InboundMiniprotocols 0.225 (TestAddr {unTestAddr = TestAddress 13}) (Bundle {withHot = WithHot [7,-24,-20,7,21,-36,-47,-2,-22], withWarm = WithWarm [16,-21,-7], withEstablished = WithEstablished [39,17,14,-37,38,34,35,-43]}),StartServer 4.1 (TestAddr {unTestAddr = TestAddress 89}) 48,InboundConnection 0.03 (TestAddr {unTestAddr = TestAddress 89}),OutboundConnection 0.012 (TestAddr {unTestAddr = TestAddress 89}),InboundMiniprotocols 1.45 (TestAddr {unTestAddr = TestAddress 89}) (Bundle {withHot = WithHot [-11,-18,32,37,10,48,-39,43,-9,11], withWarm = WithWarm [-14,39,38,-28,23,28,39,-26,46,-29,-17,18,32,14,-23,15,5,-28,-34,7,-33,46,7,-42,-38,-2,-25,-2,9,0,-27,-12,35], withEstablished = WithEstablished [-26,-23,-8,47,1,31,26,-31,25,-10,32,-10,-40,-40,-43,10,46,-43,17,-41,-4,-25,-32,33,2,25,36,-23,2,-28,9,8,38,16,-4,43,-32,-29,-36,-20]}),CloseOutboundConnection 0.024 (TestAddr {unTestAddr = TestAddress 62}),InboundMiniprotocols 0.31 (TestAddr {unTestAddr = TestAddress 89}) (Bundle {withHot = WithHot [14,12,18,8,-43,-29,37,-13,31,-10,14,25,-17,2,-44,-37,-12,10,44,36,-40,28,3], withWarm = WithWarm [22,-24,-13,-40,-31,22,23], withEstablished = WithEstablished [-18,4,35,35,30,-24,29,-48,-32,44,-46,-8,-43,45,-39,-43,-23,-31,23,31,-22,20,39,-46,-21,-37,-21,38,-33,47,28,38,-6]}),CloseOutboundConnection 0.03 (TestAddr {unTestAddr = TestAddress 13}),CloseOutboundConnection 0.02 (TestAddr {unTestAddr = TestAddress 89}),StartServer 4.6 (TestAddr {unTestAddr = TestAddress 10}) (-35),InboundConnection 3.3 (TestAddr {unTestAddr = TestAddress 10}),OutboundConnection 0.0035 (TestAddr {unTestAddr = TestAddress 10}),InboundMiniprotocols 0.45 (TestAddr {unTestAddr = TestAddress 10}) (Bundle {withHot = WithHot [41,31,-5,11,-13,-9,40], withWarm = WithWarm [27,-1,34,28,-21], withEstablished = WithEstablished [-31,-11,-43,-41,-26,24,-7,22,23,24,35,-46,-41,-27,28,-31,-35,9,20,-48,47,23,30,-26,2,7,30,7,33,-19,36,-40,-1,-27,-35,21,34,-28,-22,48]}),InboundMiniprotocols 0.42 (TestAddr {unTestAddr = TestAddress 13}) (Bundle {withHot = WithHot [46,-32,22,3,43,1,-13,-41,5,12,-24,-1,16,-43,-36,-1,-42,-29,-5,7,-27,39,9,-2,26,0,15,-40,45,-30,9,-21,-20,38,47,-36,38,-28,46,-11,20,-5,0,-27,-40,12,-45], withWarm = WithWarm [14,-22,-9,-46,46,-33,27,2,-17,-28,8,15,42,-48,-35,-12,-45,-38,47,40,45,16,41,2,-35,5,36,-26,32,26,-7,-32], withEstablished = WithEstablished [-13,38,6,43,31,31,-27,25,13,16,34]}),OutboundMiniprotocols 0.39 (TestAddr {unTestAddr = TestAddress 10}) (Bundle {withHot = WithHot [9,22,25,-38,-35,42,-24,-34,-43,-23,-47,-9,35,22,-34,10,34,21], withWarm = WithWarm [-35,-26,20,-39,-3,-29,45,27,37,11,-23,47,5,-40,-2,35,-36,36,30,-20,-41,47,-32,-38,11,-1], withEstablished = WithEstablished [48,-26,44,-23,-2,-22,43,-22,-15,-8,31,7,24,12,-2]})])

```

**Comment by coot** [comment]
Fixed in #3495.
